### ISSUE TITLE
Warning reported assigning parameter marked [DisallowNull] to object

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#40139]
No warning should be reported:
```C#
#nullable enable

using System.Diagnostics.CodeAnalysis;

class C<T>
{
    object _f;

    C([DisallowNull]T t)
    {
        _f = t; // warning CS8601: Possible null reference assignment
    }
}
```

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/36039 (Should nullability attributes affect method bodies and OHI?)
